source/MultiGrid/MultiGrid.js

```diff
@@ -0,0 +1,338 @@
+
+    if (
+      fixedRowCount !== nextProps.fixedRowCount ||
+      rowHeight !== nextProps.rowHeight
+    ) {
+      this._topGridHeight = null
+    }
+  }
+
+  render () {
+    const { height, style, width } = this.props
+    const { scrollLeft, scrollTop } = this.state
+
+    return (
+      <div
+        style={{
+          height,
+          width,
+          ...style
+        }}
+      >
+        <div
+          style={{
+            height: this._getTopGridHeight(),
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderTopLeftGrid()}
+          {this._renderTopRightGrid({ scrollLeft })}
+        </div>
+        <div
+          onScroll={this._onScrollVertical}
+          ref={(ref) => {
+            this._scrollingContainer = ref
+          }}
+          style={{
+            height: height - this._getTopGridHeight(),
+            overflowY: 'auto',
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderBottomLeftGrid({ scrollTop })}
+          {this._renderBottomRightGrid({ scrollTop })}
+        </div>
+      </div>
+    )
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState)
+  }
+
+  _cellRendererBottomLeftGrid ({ rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      rowIndex: rowIndex + fixedRowCount
+    })
+  }
+
+  _cellRendererBottomRightGrid ({ columnIndex, rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + columnIndex,
+      rowIndex: rowIndex + fixedRowCount
+    })
+  }
+
+  _cellRendererTopRightGrid ({ columnIndex, ...rest }) {
+    const { cellRenderer } = this.props
+
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + columnIndex
+    })
+  }
+
+  _columnWidthRightGrid (index) {
+    const { fixedColumnCount, columnWidth } = this.props
+
+    return columnWidth instanceof Function
+      ? columnWidth(index + fixedColumnCount)
+      : columnWidth
+  }
+
+  _getBottomGridHeight () {
+    const { height } = this.props
+
+    let topGridHeight = this._getTopGridHeight()
+
+    return height - topGridHeight
+  }
+
+  _getLeftGridWidth () {
+    const { fixedColumnCount, columnWidth } = this.props
+
+    if (this._leftGridWidth == null) {
+      if (columnWidth instanceof Function) {
+        let leftGridWidth = 0
+
+        for (let index = 0; index < fixedColumnCount; index++) {
+          leftGridWidth += columnWidth(index)
+        }
+
+        this._leftGridWidth = leftGridWidth
+      } else {
+        this._leftGridWidth = columnWidth * fixedColumnCount
+      }
+    }
+
+    return this._leftGridWidth
+  }
+
+  _getRightGridWidth () {
+    const { width } = this.props
+
+    let leftGridWidth = this._getLeftGridWidth()
+
+    return width - leftGridWidth
```

Comment:
This calculation does not consider the scrollbar width. Thus when you (or your browser) set scrollbars to be always visible the left and the right grid will overflow the total width of the table:
![image](https://cloud.githubusercontent.com/assets/8394418/20946436/f1090754-bc0a-11e6-983d-83d5312b1f93.png)
Adding scrollbar widths to this calculation is a bad idea, though, because scrollbars can have different dimensions depending on the browser.
We should probably not calculate this width manually, but rather make the right grid take up the remaining space automatically. We could probably use something like Flexbox to do this.